flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniCluster.java

```diff
@@ -438,9 +438,8 @@ protected HighAvailabilityServices createHighAvailabilityServices(Configuration
 						rpcServicesTerminationFuture,
 						this::terminateMiniClusterServices);
 
-					final CompletableFuture<Void> executorsTerminationFuture = FutureUtils.runAfterwards(
-						remainingServicesTerminationFuture,
-						() -> terminateExecutors(shutdownTimeoutMillis));
+					final CompletableFuture<Void> executorsTerminationFuture = remainingServicesTerminationFuture
```

Comment:
I think it should be `FutureUtils.composeAfterwards(remainingServicesTerminationFuture, () -> terminateExecutors(shutdownTimeoutMillis)`. Otherwise `terminateExecutors` would not be executed in case of an exceptional completion of `remainingServicesTerminationFuture`.

Reply:
Right. Updating...

Reply:
Also I update the calls to `thenCompose` with `closeMetricSystem`. Correspondingly update pull request description and commit info.